### PR TITLE
WIP: Call create api plugin in sdk command

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Call the plugin to execute the sdk command. 

**Steps to check it locally**
- Run `operator-sdk add api --api-version=cache.example.com/v1alpha1 --kind=Memcached` in a Helm project. It should create the API. 

/hold
Blocked by: https://github.com/operator-framework/operator-sdk/pull/3229

### CODE REVIEW

File: cmd/operator-sdk/add/api.go [hunk 1]
```diff
@@ -139,23 +140,30 @@ func apiRun(cmd *cobra.Command, args []string) error {
...
-			return err
+		helmPlugin := helm.Plugin{}
+		helmCreateAPIPlugin := helmPlugin.GetCreateAPIPlugin()
+
+		// The new command ask for APIVersion (e.g app.example.com/v1alpha1)
+		// Where; app == group , example.com == domain, v1alpha1 == version
+		// However, inside of the current implementation we also will break as requested by the new layout
+		// So, we are here parsing the value and adding flags for the plugin
+		api := strings.Split(apiFlags.APIVersion, "/")
+		version := api[1]
```
Comment (tengqm) [inline cmd/operator-sdk/add/api.go:151]: the APIVersion string may and may not contain a "/".

File: cmd/operator-sdk/add/api.go [hunk 2]
```diff
@@ -139,23 +140,30 @@ func apiRun(cmd *cobra.Command, args []string) error {
...
+		group := strings.Split(api[0], ".")[0]
+		domain := string(api[0][len(group)+1 : len(api[0])])
```
Comment (tengqm) [inline cmd/operator-sdk/add/api.go:153]: same here. we'd better assume the string is not of the expected format.

File: internal/plugins/helm/api.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	// Helm flags
+	//todo: Domain should be removed when the legacy layout be no longer supported
+	Domain           string
+	Group            string
+	Version          string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
```
Comment (tengqm) [inline internal/plugins/helm/api.go:52]: Rename HelmFoo to Foo for simplification? Looks like we are inside an struct which is meant to be used only inside the module.
Also pls add some comments to the fields? For example, ChartRef is a confusing field without a comment.

File: internal/plugins/helm/api.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		_, err := sdkscaffold.NewResource(path.Join(p.Group+"."+p.Domain, p.Version), p.Kind)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *createAPIPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
+	// Check if the project name is a valid namespace according to k8s
```
Comment (tengqm) [inline internal/plugins/helm/api.go:200]: remove line 200

**Comment by camilamacedo86** [comment]
It will change. Closing for now but I might will re-open this one.

File: cmd/operator-sdk/add/api.go (hunk 1 above)
Comment (camilamacedo86) [reply cmd/operator-sdk/add/api.go:151]: it is mandatory

File: cmd/operator-sdk/add/api.go (hunk 2 above)
Comment (camilamacedo86) [reply cmd/operator-sdk/add/api.go:153]: Ditto.

**Pull request closed without merging**